src/js/export-favicon.js.rst

```diff
@@ -53,46 +54,69 @@ Start by ensuring a single layer has been selected as the favicon export source.
     export default function (context) {
       const sketch = context.api();
 
+      const errors = [];
       const layers = sketch.selectedDocument.selectedLayers;
 
       if (layers.length === 0) {
-        sketch.alert('Select a layer to export as favicon.', 'Error');
-      } else if (layers.length > 1) {
-        sketch.alert('Select a single layer to export as favicon.', 'Error');
-      } else {
+        errors.push('Select a layer to export as favicon.');
+      }
+
+      if (layers.length > 1) {
+        errors.push('Select a single layer to export as favicon.');
+      }
+
+      layers.iterate(layer => {
+        const bounds = layer.sketchObject.absoluteRect();
+
+        if (bounds.width() !== bounds.height()) {
+          errors.push('Select a square layer to export as favicon.');
+        }
+        if (bounds.width() < 256) {
+          errors.push('Select a layer at least 256x256 to export as favicon.');
+        }
+      });
+
+None of these errors are recoverable so error out. Only need to specify a single
+error cause for this.
+
+.. code-block:: javascript
+
+      if (errors.length > 0) {
+        sketch.alert(errors[0], 'Error');
```

Comment:
Wouldn't it be nicer to tell the user about all the collected errors?

Reply:
Always a tricky call this one. A list of errors can be quite intimidating on one hand. Fixing simple issues one at a time is annoying on the other hand.

Didn't think much about this beforehand but looking at it the selection count validation could be a nuisance. e.g. more than two layers selected and size/shape errors in each, say. In fairness, the selection count validation would want to go first and the other validations cascade after it but then the validation isn't so straight line.

I think we can be okay with letting the user fix one thing at a time here. Any more validations and we might want to rework the validation errors to be more helpful, e.g. combine the size/shape validation.